nautobot/extras/models/metadata.py

```diff
@@ -135,3 +141,249 @@ def save(self, *args, **kwargs):
+                    if not isinstance(value, datetime):
+                        try:
+                            datetime.strptime(value, "%Y-%m-%d %H:%M:%S")
+                        except TypeError:
+                            raise ValidationError("Value must be a datetime or str object.")
+                        except ValueError:
+                            raise ValidationError("Datetime values must be in the format YYYY-MM-DD HH:MM:SS.")
+
+                # Validate selected choice
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_SELECT:
+                    if value not in self.metadata_type.choices.values_list("value", flat=True):
+                        raise ValidationError(
+                            f"Invalid choice ({value}). Available choices are: {', '.join(self.metadata_type.choices.values_list('value', flat=True))}"
+                        )
+
+                elif metadata_type_data_type == MetadataTypeDataTypeChoices.TYPE_MULTISELECT:
+                    if isinstance(value, str):
+                        value = value.split(",")
+                    if not set(value).issubset(self.metadata_type.choices.values_list("value", flat=True)):
+                        raise ValidationError(
+                            f"Invalid choice(s) ({value}). Available choices are: {', '.join(self.metadata_type.choices.values_list('value', flat=True))}"
+                        )
+            else:
+                raise ValidationError(
+                    f"value is a required field that cannot be empty for metadata type {metadata_type_data_type}."
+                )
+
+        if self.present_in_database:
+            # Check immutable fields
+            database_object = self.__class__.objects.get(pk=self.pk)
+            if self.metadata_type != database_object.metadata_type:
+                raise ValidationError({"metadata_type": "Cannot be changed once created"})
+
+        # Check if there is any intersections of scoped_fields of ObjectMetadata instances in the database.
+        object_metadata_scoped_fields = (
+            ObjectMetadata.objects.filter(
+                metadata_type=self.metadata_type,
+                assigned_object_type=self.assigned_object_type,
+                assigned_object_id=self.assigned_object_id,
+            )
+            .exclude(pk=self.pk)
+            .values_list("scoped_fields", flat=True)
+        )
+        duplicate_scoped_fields_list = set([])
+
+        for scoped_fields in object_metadata_scoped_fields:
+            duplicate_scoped_fields_list |= set(self.scoped_fields).intersection(scoped_fields)
+
+        if duplicate_scoped_fields_list:
+            raise ValidationError(
+                {
+                    "scoped_fields": f"There are other Object Metadata instances of metadata type {self.metadata_type} scoping the same fields {duplicate_scoped_fields_list} for {self.assigned_object}"
+                }
+            )
```

Comment:
IIRC, per the design, a scoped_fields of `[]` implies "all fields on this record"? If that's the case, then should `[]` be considered to conflict with *any* other metadata of this type on this record?